### PR TITLE
Ensure cache encryption is not reversible

### PR DESCRIPTION
**Pull request by morgante**
Fixes https://github.com/braintrustdata/braintrust-proxy/issues/44

As demonstrated in the test script, the previous encryption strategy was reversible as long as you had the OpenAI key used. At least for my use case, this violates the guarantees we try to give customers of not retaining copies of their data.

This switches it to use the request data itself as the encryption key, such that you can't decrypt the cache unless you actually have the original user request in hand.

Note: I don't have a Vercel or Node server handy, so this has only been tested with Wrangler/Cloudflare.

### CODE REVIEW

File: apis/cloudflare/src/encrypt.ts
Comment (ankrgyl) [inline apis/cloudflare/src/encrypt.ts:1]: nice :)

File: packages/proxy/test/decrypt.ts
Comment (ankrgyl) [inline packages/proxy/test/decrypt.ts:1]: I think we should either remove this file or introduce `vitest`/`jest` into the repo and make this a test. Happy to help with the latter if you're open to it, or I can clean it up in a follow up change.

File: packages/proxy/src/proxy.ts
Comment (ankrgyl) [inline packages/proxy/src/proxy.ts:176]: This logic looks pretty straightforward to me.

File: packages/proxy/src/proxy.ts
Comment (ankrgyl) [inline packages/proxy/src/proxy.ts:183]: Did you run the pre-commit hooks? You need to install https://pre-commit.com/ and then just run

```
pre-commit 
```

**Review by ankrgyl — COMMENTED** [review]
Thank you immensely for contributing this! I see what you mean about wanting to encrypt using the body as well. I will run our internal tests on this and report if there are any issues.

File: packages/proxy/test/decrypt.ts
Comment (morgante) [reply packages/proxy/test/decrypt.ts]: Ok, I removed it. Feel free to add as a test later if you decide to incorporate vitest.

**Review by ankrgyl — APPROVED** [review]
Thank you! This is looking good. We need to resolve some stuff with our internal tests, because we cache a bunch of things. Once we figure this out, I'll land this and deploy!

Let me know if you are blocked on this, and if so, we'll make sure to move things along faster.

File: packages/proxy/src/proxy.ts
Comment (morgante) [reply packages/proxy/src/proxy.ts]: I didn't see a format or prettier command, but ran pre-commit now.

**Comment by morgante** [comment]
> Let me know if you are blocked on this, and if so, we'll make sure to move things along faster.

Not blocked, I've already deployed a version of this for our usage.

**Comment by ankrgyl** [comment]
Ok we are ready to land this! Thanks again for the contribution

**Pull request merged**
